client/state/query-client.ts

```diff
@@ -26,6 +50,25 @@ export async function createQueryClient( userId: number | undefined ): Promise<
 			queryClient,
 			persistor,
 			maxAge: MAX_AGE,
+			dehydrateOptions: {
+				shouldDehydrateQuery: ( query ) => {
+					if ( query.state.status !== 'success' ) {
+						return false;
+					}
+
+					const shouldPersist = query.meta?.[ SHOULD_PERSIST_QUERY_KEY ];
```

Comment:
Won't this interfere with the Redux `shouldPersist` import from above?

Reply:
Yeah! We could rename it to `shouldPersistQuery`. What do you think?

Reply:
Sounds good to me 👍